### ISSUE TITLE
JAR size has increased due to included fonts

### ISSUE THREAD

**Issue by lapo-luchini** [LibrePDF/OpenPDF#277]
JAR size has increased a lot:

    1899402 openpdf-1.1.0.jar
    4243560 openpdf-1.3.11.jar
    1898896 openpdf-1.3.11-without-TTFs.jar

mostly thanks to bc98cd8f5137b6dc5cadc65cf80c3ec37e62e2d0 adding Liberation fonts.
Those seems like very useful in other use-cases but are they necessary in basic parsing / changing in a non-graphical way / digital signature of PDFs?
Would it be possible to include the fonts in an external JAR dependency instead of the main JAR, so that it could be excluded as a dependency in a project that doesn't need them?
I'm just wondering, if it's not worth the hassle, no problem.

**Comment by asturio** [comment]
Yes, that is true. The fonts are quite big. They are the only Unicode-Fonts included in the package, as many user had problems generating PDF in non English languages. And the missing fonts is the problem. The old included fonts were really ancient and supported only a small set of characters.

I think actually the user should be responsible for using the correct font, but many just don't do it, so the necessity of including some default set.

The choice was quite simple: Unicode, LGPL compatible, and with metrics compatible to the standard, common fonts like Arial and Times.

We could make a separate optional package containing the Fonts, if this is requested some more times.

So please, vote this up, if you think the ~2.4 Mb more because of the fonts is really an issue.

**Comment by exuvo** [comment]
I also see the filesize increase as an issue. We deploy many tomcat webapplications and an 2.4MB lib size increase would multiply many times across all the webapplications leading to a measurable increase in disk usage and redeployment times. Unfortunately we are not able to use filesystem deduplication at this time and even with it we still have to transfer it to a handful of servers some remote with slower connections.

**Comment by asturio** [comment]
Ok, we could make a separate package with only the added fonts. So if you do not need UTF-8 support, or have your own fonts, you won't need the extra 2.4 Mb...

**Comment by lapo-luchini** [comment]
I'd say this has been solved in 1.3.21:

```
% ll */*.jar
-rwxr--r-- 1 lapo lapo 1880897 giu 23 18:17 1.2.17/openpdf-1.2.17.jar
-rwxr--r-- 1 lapo lapo 4243560 lug 30 15:12 1.3.11/openpdf-1.3.11.jar
-rwxr--r-- 1 lapo lapo 4251280 giu 23 18:17 1.3.19/openpdf-1.3.19.jar
-rwxr--r-- 1 lapo lapo 4253615 giu 25 17:25 1.3.20/openpdf-1.3.20.jar
-rwxr--r-- 1 lapo lapo 2116586 set 23 11:19 1.3.21/openpdf-1.3.21.jar
```

**Comment by lapo-luchini** [comment]
(still increased 1898896 → 2116586, but I'd say that's minor)
